src/topology/algebra/order/left_right.lean

```diff
@@ -61,3 +86,260 @@ lemma continuous_at_iff_continuous_left'_right' {a : α} {f : α → β} :
   continuous_at f a ↔ continuous_within_at f (Iio a) a ∧ continuous_within_at f (Ioi a) a :=
 by rw [continuous_within_at_Ioi_iff_Ici, continuous_within_at_Iio_iff_Iic,
   continuous_at_iff_continuous_left_right]
+
+end topological_space
+
+section left_right_lim
+
+section
+
+variables {α β : Type*} [linear_order α] [topological_space β]
+
+/-- Let `f : α → β` be a function from a linear order `α` to a topological_space `β`, and
+let `a : α`. The limit strictly to the left of `f` at `a`, denoted with `left_lim f a`, is defined
+by using the order topology on `α`. If `a` is isolated to its left, we use `f a` instead to
+guarantee a good behavior in most cases. -/
+@[irreducible] noncomputable def function.left_lim (f : α → β) (a : α) : β :=
+begin
+  classical,
+  haveI : nonempty β := ⟨f a⟩,
+  letI : topological_space α := preorder.topology α,
+  exact if (𝓝[<] a) = ⊥ then f a else lim (𝓝[<] a) f
+end
+
+/-- Let `f : α → β` be a function from a linear order `α` to a topological_space `β`, and
+let `a : α`. The limit strictly to the right of `f` at `a`, denoted with `right_lim f a`, is defined
+by using the order topology on `α`. If `a` is isolated to its left, we use `f a` instead to
+guarantee a good behavior in most cases. -/
+noncomputable def function.right_lim (f : α → β) (a : α) : β :=
+@function.left_lim αᵒᵈ β  _ _ f a
+
+open function
+
+lemma left_lim_eq_of_ne_bot [hα : topological_space α] [h'α : order_topology α]
+  (f : α → β) {a : α} (h : 𝓝[<] a ≠ ⊥) :
+  left_lim f a = @lim _ _ _ ⟨f a⟩ (𝓝[<] a) f :=
+begin
+  rw [h'α.topology_eq_generate_intervals] at h ⊢,
+  simp [left_lim, ite_eq_right_iff, h],
+end
+
+lemma left_lim_eq_of_eq_bot [hα : topological_space α] [h'α : order_topology α]
+  (f : α → β) {a : α} (h : 𝓝[<] a = ⊥) :
+  left_lim f a = f a :=
+begin
+  rw [h'α.topology_eq_generate_intervals] at h,
+  simp [left_lim, ite_eq_left_iff, h],
+end
+
+end
+
+open function
+
+namespace monotone
+
+variables {α β : Type*} [linear_order α] [conditionally_complete_linear_order β]
+[topological_space β] [order_topology β]
+{f : α → β} (hf : monotone f)  {x y : α}
+include hf
+
+lemma left_lim_eq_Sup [topological_space α] [order_topology α] (h : 𝓝[<] x ≠ ⊥) :
+  left_lim f x = Sup (f '' (Iio x)) :=
+begin
+  haveI : ne_bot (𝓝[<] x) := ne_bot_iff.2 h,
+  rw left_lim_eq_of_ne_bot f h,
+  exact (hf.tendsto_nhds_within_Iio x).lim_eq,
+end
+
+lemma left_lim_le (h : x ≤ y) : left_lim f x ≤ f y :=
+begin
+  letI : topological_space α := preorder.topology α,
+  haveI : order_topology α := ⟨rfl⟩,
```

Comment:
Should we add a tactic similar to `borelize` (not sure)?

Reply:
I don't think so. The need to do that is only because of my slightly weird definition of `left_lim` (where I didn't want to assume a topology on the source space, because it's not necessary), but it's a very local phenomenon.